openpathsampling/experimental/simstore/storable_functions.py

```diff
@@ -317,7 +325,7 @@ def mode(self):
 
     @mode.setter
     def mode(self, value):
-        allowed_values = ['no-caching', 'analysis', 'production']
+        allowed_values = list(self._modes.keys())
         if value not in allowed_values:
```

Comment:
This is unnecessary (`in` always checks against `keys()` for `dict`s) :
```python
>>> a = {"a": 1, "b": 2, "c":3}
>>> "a" in a
True
>>> 1 in a
False
```
```suggestion
        if value not in self._modes:
```

Reply:
It's necessary because the error message reports `allowed_values`. I thought it was more readable to set it here and re-use it than to have the error message require `list(self._modes)` (although it could be `list(self._modes)` instead of `list(self._modes.keys())`

Reply:
+1, Missed the error (collapsed in the review)
```suggestion
        allowed_values = list(self._modes)
        if value not in allowed_values:
```